From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TheDBP <TheDBP@users.noreply.github.com>
Date: Sun, 20 Sep 2026 09:35:43 -0400
Subject: [PATCH] bonito: drop the HIDL power.stats 1.0 HAL for now

android.hardware.power.stats@1.0 is in no framework compatibility matrix
from level 7 up, so check_vintf's unused-HAL check rejects the instance:

  android.hardware.power.stats@1.0::IPowerStats/default

Stop shipping android.hardware.power.stats@1.0-service.pixel and remove
its manifest entry. The powerstats sources and sepolicy stay in place for
a later AIDL power.stats port (coral did the same, then ported later).
---
 device.mk    | 4 ----
 manifest.xml | 9 ---------
 2 files changed, 13 deletions(-)

diff --git a/device.mk b/device.mk
index 3a4b3a20..156b67d0 100644
--- a/device.mk
+++ b/device.mk
@@ -206,10 +206,6 @@ PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.hardware.strongbox_keystore.xml:$(TARGET_COPY_OUT_VENDOR)/etc/permissions/android.hardware.strongbox_keystore.xml \
     frameworks/native/data/etc/android.software.ipsec_tunnels.xml:$(TARGET_COPY_OUT_VENDOR)/etc/permissions/android.software.ipsec_tunnels.xml \
 
-# powerstats HAL
-PRODUCT_PACKAGES += \
-    android.hardware.power.stats@1.0-service.pixel
-
 PRODUCT_COPY_FILES += \
     $(LOCAL_PATH)/powerhint.json:$(TARGET_COPY_OUT_VENDOR)/etc/powerhint.json
 
diff --git a/manifest.xml b/manifest.xml
index 9f443778..76e3d853 100644
--- a/manifest.xml
+++ b/manifest.xml
@@ -120,15 +120,6 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="hidl">
-        <name>android.hardware.power.stats</name>
-        <transport>hwbinder</transport>
-        <version>1.0</version>
-        <interface>
-            <name>IPowerStats</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="hidl">
         <name>vendor.display.config</name>
         <transport>hwbinder</transport>
